python/tvm/relax/frontend/legalizer.py

```diff
@@ -0,0 +1,136 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import math
+
+import tvm
+from tvm import relax, te, topi
+from tvm.ir.module import IRModule
+from tvm.relax.block_builder import BlockBuilder
+
+
+@relax.expr_functor.mutator
+class OperatorLegalizer(relax.PyExprMutator):
+    def __init__(self, mod: IRModule) -> None:
+        super().__init__(mod)
+        self.mod_ = mod
+
+    def _convert_op(self, call: relax.Call) -> relax.Call:
+        bb: BlockBuilder = self.builder_
+        op = call.op
+        attrs = call.attrs
+        args = call.args
+
+        if op.name == "relax.nn.conv2d":
```

Comment:
use a dispatch table(instead of if then else) so it is extensible, see https://mlc.ai/chapter_integration/index.html

Reply:
Right it was just updated.